shared/config/config.go

```diff
@@ -30,10 +30,15 @@ type Pre2PConfig struct {
 }
 
 type P2PConfig struct {
-	Protocol   string          `json:"protocol"`
-	Address    pcrypto.Address `json:"address"`
-	ExternalIp string          `json:"external_ip"`
-	Peers      []string        `json:"peers"`
+	Protocol         string          `json:"protocol"`
+	Address          pcrypto.Address `json:"address"`
+	ExternalIp       string          `json:"external_ip"`
+	Peers            []string        `json:"peers"`
+	MaxInbound       uint32          `json:max_inbound`
+	MaxOutbound      uint32          `json:max_outbound`
+	BufferSize       uint            `json:connection_buffer_size`
```

Comment:
Inbound/Outbound maxes are usually set around 128, 256 and what not, and since we don't want to limit the possible maxes (_which will heavily depend on the machines capacity_) I've went with 32 (_instead of 8 which will result in 256_). To come to think of it, `Uint16` is enough.

The other values such as buffer size and timeouts are freely set values and are convention based, so the only part that's required to be specified about their type is that they are positive, and they are integers.